### PR TITLE
add context

### PR DESCRIPTION
**Pull request by jmank88**
https://smartcontract-it.atlassian.net/browse/BCF-2887

Requires:
- https://github.com/smartcontractkit/chainlink-common/pull/490
- https://github.com/smartcontractkit/tdh2/pull/37

Supports:
- https://github.com/smartcontractkit/chainlink/pull/14011

core ref: f4c9b9edec10085f629888ac6b50efca3fcd054b

### CODE REVIEW

File: integration-tests/go.mod [hunk 1]
```diff
@@ -2,6 +2,10 @@ module github.com/smartcontractkit/chainlink-solana/integration-tests
 
 go 1.21.7
 
+replace github.com/smartcontractkit/tdh2/go/ocr2/decryptionplugin => github.com/jmank88/tdh2/go/ocr2/decryptionplugin v0.0.0-20240321204202-bf6e0344b64b
```
Comment (krehermann) [inline integration-tests/go.mod]: what is this?

File: integration-tests/go.mod [hunk 2]
```diff
@@ -2,6 +2,10 @@ module github.com/smartcontractkit/chainlink-solana/integration-tests
...
+
+replace github.com/smartcontractkit/tdh2/go/tdh2 => github.com/jmank88/tdh2/go/tdh2 v0.0.0-20240321204202-bf6e0344b64b
```
Comment (krehermann) [inline integration-tests/go.mod]: and this?

File: integration-tests/go.mod (hunk 1 above)
Comment (jmank88) [reply integration-tests/go.mod]: I didn't have auth to push a branch to tdh2 so this was required to use my personal fork of that repo until the PR is merged.

File: integration-tests/go.mod [hunk 3]
```diff
@@ -56,6 +60,8 @@ require (
 	github.com/Masterminds/sprig/v3 v3.2.3 // indirect
 	github.com/Microsoft/go-winio v0.6.1 // indirect
 	github.com/Microsoft/hcsshim v0.11.4 // indirect
+	github.com/NethermindEth/juno v0.3.1 // indirect
+	github.com/NethermindEth/starknet.go v0.6.1-0.20231218140327-915109ab5bc1 // indirect
```
Comment (aalu1418) [inline integration-tests/go.mod]: probably unavoidable at the moment - but now we have starknet deps in solana integration tests 😢

File: pkg/solana/monitor/balance.go [hunk 4]
```diff
@@ -62,14 +62,14 @@ func (b *balanceMonitor) Name() string {
 }
 
 func (b *balanceMonitor) Start(context.Context) error {
-	return b.StartOnce("SolanaBalanceMonitor", func() error {
+	return b.StartOnce("BalanceMonitor", func() error {
```
Comment (aalu1418) [inline pkg/solana/monitor/balance.go:67]: just double checking - is there other indication that this is a solana balance monitor? (with and without loop)

File: pkg/solana/relay.go [hunk 5]
```diff
@@ -50,51 +45,61 @@ func (r *Relayer) Name() string {
...
+func (r *Relayer) GetChainStatus(ctx context.Context) (relaytypes.ChainStatus, error) {
+	return r.chain.GetChainStatus(ctx)
+}
+
+func (r *Relayer) ListNodeStatuses(ctx context.Context, pageSize int32, pageToken string) (stats []relaytypes.NodeStatus, nextPageToken string, total int, err error) {
+	return r.chain.ListNodeStatuses(ctx, pageSize, pageToken)
+}
+
+func (r *Relayer) Transact(ctx context.Context, from, to string, amount *big.Int, balanceCheck bool) error {
+	return r.chain.Transact(ctx, from, to, amount, balanceCheck)
```
Comment (aalu1418) [inline pkg/solana/relay.go:96]: breaking interface changes?

will probably want to get everything running between this PR and core PR before merging this one (trying to reduce the amount of time something would be broken)

**Review by aalu1418 — COMMENTED** [review]
overall looks good to me!

File: pkg/solana/monitor/balance.go (hunk 4 above)
Comment (jmank88) [reply pkg/solana/monitor/balance.go:67]: Yeah these names don't end up being used without additional context.

File: integration-tests/go.mod (hunk 3 above)
Comment (jmank88) [reply integration-tests/go.mod]: :shrug: 
Something to do with core integration-tests/ importing the chain family repos? I'm not sure.

File: pkg/solana/relay.go (hunk 5 above)
Comment (jmank88) [reply pkg/solana/relay.go:96]: Yeah I wanted to get these clean and collapse them all quickly, but we're clogged up on tdh2 now unfortunately. I will wait to open this until it is ready.

File: integration-tests/go.mod (hunk 3 above)
Comment (aalu1418) [reply integration-tests/go.mod]: yeah it is - not much we can do about it at the moment tho

**Review by krehermann — APPROVED** [review]

**Review by aalu1418 — APPROVED** [review]

**Pull request merged**